### PR TITLE
Automate android build

### LINKED ISSUE #4707: Automate Android Build
## The suggestion

Use the NZ Mac Mini to automatically create an android build when a release tag being create.
Alternatively we could do it in Jenkins, if that made it easier somehow?

### Why should we invest time in this?

Would save time for release team creating a android release for every release
Would ensure the full build process was defined in code, reducing the reliance on sharing knowledge & keys for production builds.

### Are there any risks associated with this change?

Automating a process may take more time than running it manually.

### How much effort is required?

Feels like a 2-4 day job? Could go faster potentially but I think it's worth spending a few days on it if needed.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #4707 

# 👩🏻‍💻 What does this PR do?
Adds the android build steps to the windows batch files
Strictly speaking this isn't a windows function, but I've included it in the batch scripts so that the commands are in version control.

The configuration on jenkins is:
* download the [ndk](https://developer.android.com/ndk/downloads) and sdk [command line tools](https://developer.android.com/studio#command-line-tools-only)
* Set up the following folder structure in the root of C:
  * android
    * sdk
      * cmdline-tools
        * latest ( the command line tools download, extracted, with the unzipped directory renamed from `cmdline-tools` to `latest` and copied to `cmdline-tools` )
    * ndk ( the downloaded ndk is extracted here, note: on jenkins I kept the version in the name too, so this is ndk-r27) 
* add an environment variable `ANDROID_HOME=C:\android\sdk`
* add an environment variable `NDK_BIN=c:\android\ndk` (note: on jenkins I kept the version in the name too, so this is ndk-r27)
* copy the `local.properties' and 'release.keystore` files to `c:\android` as well - the batch script copies them from there

The script also renames the apk to be in the same format as the other installers, viz: `Open_mSupply_Android_%versionTag%.apk`

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Switch to this branch and create a tag starting with 'v', push that tag to origin and wait for jenkins to do stuff

```
git tag v[Something or other]
git push origin tag v[Something or other]
```

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: build/windows/omsupply-build.bat [hunk 1]
```diff
@@ -17,7 +17,12 @@ start /wait /b build\windows\omsupply-prepare.bat
 @if %errorlevel% neq 0 exit /b %errorlevel%
 
 @ECHO ##### Building omsupply for sqlite #####
-cd server && cargo build --release --bin omsupply_service && copy "target\release\omsupply_service.exe" "..\omSupply\Server\omSupply-sqlite.exe"
+@cd server
```
Comment (mark-prins) [inline build/windows/omsupply-build.bat:20]: the rearranging isn't necessary for the android build, it just shifts the sqlite cargo compilation to be together, in the hope that it reduces some of the compilation time.

File: build/windows/omsupply-build.bat [hunk 2]
```diff
@@ -29,14 +34,19 @@ cargo build --release --bin remote_server_cli && copy "target\release\remote_ser
 cargo build --release --bin remote_server_cli --features postgres && copy "target\release\remote_server_cli.exe" "..\omSupply\Server\omSupply-cli-postgres.exe"
 @if %errorlevel% neq 0 exit /b %errorlevel%
 
-@ECHO ##### Building omSupply cli #####
+@ECHO ##### Building connection test utility #####
```
Comment (mark-prins) [inline build/windows/omsupply-build.bat:37]: copypaste correction

**Comment by mark-prins** [comment]
For testing: [this build](https://jenkins.msupply.org:8443/job/omSupplyMain%20-%20Installers/lastSuccessfulBuild/) worked 😉

**Review by lache-melvin — APPROVED** [review]
Looks good to me! Thanks for all the tinkering Mark 😁

**Pull request merged**